fathom/src/surface/distillation.rs

```diff
@@ -86,13 +87,14 @@ impl<'interner, 'arena, 'env> Context<'interner, 'arena, 'env> {
         &mut self,
         number: T,
         style: UIntStyle,
-    ) -> Term<'arena, ()> {
+        span: Span,
+    ) -> Term<'arena, Span> {
```

Comment:
We might be able to unparameterise the surface terms at this point 🤔

Reply:
I started doing this but I'm not sure I like the end result. Since we need to be able to produce `Terms` with `Spans` I assume that the `Range` parameter would be replaced with `Span`. However that means that all the places where the parameter was `ByteRange` now accept/need to handle `Span::Empty`, which feels less accurate (Since Term<ByteRange> always has a file id and range). I think for now it might be worth leaving this as-is. Happy to debate the pros and cons though.

Reply:
Tbh I'm a little uncomfortable returning `Spans` here as I kind of consider distilled code to be generated. Maybe in the future we could add some ‘generated span’ variant that points back to the original source span. 🤔

Reply:
Thanks for noticing and highlighting it. I've reworked the commits to not populate distilled `Terms` with a `Span`. It cleans up a lot of `TODO`/`FIXME` questions I had and doesn't impact the diagnostics that can be generated.